### ISSUE TITLE
Remove default for allowed wildcard imports in `ktlint_official` codestyle

### ISSUE THREAD

**Issue by paul-dingemans** [pinterest/ktlint#1797]
>Expected Behavior
> When java.util.* is included as a wildcard import, it should trigger the warning for no-wildcard-imports.
> ...
> Given that, I think it makes sense that ktlint should stop including those two paths as default wildcard exceptions.

_Originally posted by @greg-patterson in https://github.com/pinterest/ktlint/issues/1792#issuecomment-1407098815_

**Comment by paul-dingemans** [comment]
The feature to respect setting ij_kotlin_packages_to_use_import_on_demand  was implemented not too long ago in ktlint 0.45.x via https://github.com/pinterest/ktlint/pull/1272. The reason for adding the default values is that they seems to be hard-coded in IntelliJ IDEA as well (https://github.com/JetBrains/kotlin/blob/ffdab473e28d0d872136b910eb2e0f4beea2e19c/idea/formatter/src/org/jetbrains/kotlin/idea/core/formatter/KotlinCodeStyleSettings.java#L81-L82).

If the property is not set explicitly, IntelliJ IDEA allows imports java.util.* and kotlinx.android.synthetic.*. If ktlint would not do the same, this would result in a conclict between ktlint and IntelliJ IDEA. Lots of users of ktlint dislike that.

In the new codestyle `ktlint_official` the default value for allowed wildcard imports shoulbe be cleared.

**Comment by paul-dingemans** [comment]
Removing the default value for code `ktlint_official` resulted in ktlint to report wildcard imports for `java.util.*` in case that property `ij_kotlin_packages_to_use_import_on_demand` is not set. But IntelliJ IDEA interpret the absence of the property differently and stil add the the `java.util.*` import.

To keep KtLint and IntelliJ IDEA formatter in sync, it is needed to set `.editorconfig` property below explitcily:
```
ij_kotlin_packages_to_use_import_on_demand = unset
```
